pkgs/build-support/bintools-wrapper/default.nix

```diff
@@ -145,7 +145,7 @@ stdenv.mkDerivation {
         ln -s $ldPath/${targetPrefix}as $out/bin/${targetPrefix}as
       fi
 
-    '' + (if !useMacosReexportHack then ''
+    '' + (if !useMacosReexportHack then optionalString (!targetPlatform.isDarwin) ''
```

Comment:
I don't think this is right. We need to wrap ld in binutils to get the right ldflags.

Reply:
Hm. I've been noodling about this one a while. [This comment on darwin's binutils](https://github.com/nixos/nixpkgs/blob/4073a8b20b3d65dbde91b746cceb61d189bba366/pkgs/os-specific/darwin/binutils/default.nix#L26) says it though: binutils on darwin doesn't provide `ld`, and shipping an `ld` in binutils-wrapper means anything that has the wrapper on `$PATH` will invoke a wrapper that then blindly tries to invoke an unwrapped binary that doesn't exist: The result is that on Darwin, you get "No such file or directory" invoking `ld` and get very confused.

Reply:
@matthewbauer is right. This wrapper wraps that direction which *does* provide `ld` by getting it from Apple's `cctools`.

Reply:
I think I didn't express this clearly enough: The binutils wrapper for `ld` on darwin, as installed in the unpatched state, is broken:

``` sh
# Running in nixpkgs with this change backed out:
$ nix-build . -A binutils
these paths will be fetched (7.63 MiB download, 47.08 MiB unpacked):
  /nix/store/4z8l6d7p2l31xfwypg2i7ykjar7r3n0x-gettext-0.21
  /nix/store/6an46yg377ccqs5w9na64xmdcp6f2ic4-coreutils-8.31
  /nix/store/8p0g4jqxq1hwr0417miw5chdwkwxqlxn-libc++-7.1.0
  /nix/store/g0cpzdibxd83lzlcyl16zgwa0f95lp4y-libc++abi-7.1.0
  /nix/store/jh5jyg67b9frbvpc08jziqf5n2mnkx8v-libiconv-osx-10.12.6
  /nix/store/k31hbn227zp0pp21lygbh73p1111da6m-binutils-2.31.1
  /nix/store/lnxj6hqrwvi9q55lzk6c5mgy8aw27zq5-zlib-1.2.11
  /nix/store/nb9givrwj0m1jbjf65x4g4i6y1n7pjf0-Libsystem-osx-10.12.6
  /nix/store/pyw7sdlm0pc8a9pl4wfv6bn40igy6c1n-gmp-6.2.0
  /nix/store/qkw4ib1kapm28avkz902p8vicpglg8rv-expand-response-params
  /nix/store/r5p4a9xaa7hf3pdal0zkhc23m609shqn-bash-4.4-p23
  /nix/store/rkca4cz1caays8yc7nfq5c0b3pg97y06-binutils-wrapper-2.31.1
copying path '/nix/store/nb9givrwj0m1jbjf65x4g4i6y1n7pjf0-Libsystem-osx-10.12.6' from 'https://cache.nixos.org'...
copying path '/nix/store/r5p4a9xaa7hf3pdal0zkhc23m609shqn-bash-4.4-p23' from 'https://cache.nixos.org'...
copying path '/nix/store/qkw4ib1kapm28avkz902p8vicpglg8rv-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/g0cpzdibxd83lzlcyl16zgwa0f95lp4y-libc++abi-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/jh5jyg67b9frbvpc08jziqf5n2mnkx8v-libiconv-osx-10.12.6' from 'https://cache.nixos.org'...
copying path '/nix/store/8p0g4jqxq1hwr0417miw5chdwkwxqlxn-libc++-7.1.0' from 'https://cache.nixos.org'...
copying path '/nix/store/lnxj6hqrwvi9q55lzk6c5mgy8aw27zq5-zlib-1.2.11' from 'https://cache.nixos.org'...
copying path '/nix/store/4z8l6d7p2l31xfwypg2i7ykjar7r3n0x-gettext-0.21' from 'https://cache.nixos.org'...
copying path '/nix/store/pyw7sdlm0pc8a9pl4wfv6bn40igy6c1n-gmp-6.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/k31hbn227zp0pp21lygbh73p1111da6m-binutils-2.31.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6an46yg377ccqs5w9na64xmdcp6f2ic4-coreutils-8.31' from 'https://cache.nixos.org'...
copying path '/nix/store/rkca4cz1caays8yc7nfq5c0b3pg97y06-binutils-wrapper-2.31.1' from 'https://cache.nixos.org'...
/nix/store/rkca4cz1caays8yc7nfq5c0b3pg97y06-binutils-wrapper-2.31.1
$ ./result/bin/ld
./result/bin/ld: line 207: /nix/store/k31hbn227zp0pp21lygbh73p1111da6m-binutils-2.31.1/bin/ld: No such file or directory
```

That is, on Darwin it installs a wrapper script that clutters up $PATH but doesn't operate correctly because the thing it wraps doesn't exist.

I'm happy to back this diff out, but I do think binutils as installed on Darwin results in a misleading wrapper, which this change was mean to address.

Reply:
OK yeah that makes sense. `bintutils` is different than `stdenv.cc.bintools`, and the latter what what I was talking about and the the thing actually used on Darwin. We might in thte future make the script general robust to missing stuff, in which case `binutils` wouldn't have broken wrapper but just would be missing the things provided by `cctools`.

Reply:
To be clear, I mean what you observed makes senses. You should still revert this line like Matt said. We'd ned to do something else to fixed wrapped plan GNU binutils anyways, and that's best done in a later commit.